### PR TITLE
[TASK] Change markup

### PR DESCRIPTION
**Pull request by NeoBlack**
Fixes: #54 #58

### CODE REVIEW

File: Build/Vue/src/components/AllSelector/index.tsx [hunk 1]
```diff
@@ -26,12 +26,14 @@ export default class AllSelector extends Vue {
     }
 
     private render(): VNode {
-        // fix me, I'm ugly
         return (
-            <a href='#' onClick={(event: Event) => this.toggleSelect(event, this.listOfIdentifiers)} class='btn btn-sm btn-default'>
-                <i class='fa fa-check-square' v-show={this.isSelected} />
-                <i class='fa fa-square-o' v-show={!this.isSelected} />
-            </a>
+            <span class='component-datatable-checkbox'>
```
Comment (benjaminkott) [inline Build/Vue/src/components/AllSelector/index.tsx]: As discussed in the call earlier, we should refine the naming scheme here to be able to reuse it later in different places. Please change all `component-datatable-checkbox` to `component-checkbox`. Sorry, updated also the markup in the issue.

File: Build/Vue/src/components/AllSelector/index.tsx [hunk 2]
```diff
@@ -26,12 +26,14 @@ export default class AllSelector extends Vue {
...
+                <input class='component-datatable-checkbox-input' type='checkbox' id='component-datatable-selectall-123523356'
+                  value='1' onClick={(event: Event) => this.toggleSelect(event, this.listOfIdentifiers)} />
+                <label class='component-datatable-checkbox-label' for='component-datatable-selectall-123523356'>
+                    <span class='component-datatable-visually-hidden'>{this.isSelected ? 'deselect' : 'select'} all elements</span>
```
Comment (benjaminkott) [inline Build/Vue/src/components/AllSelector/index.tsx]: Same goes for `component-datatable-visually-hidden` - we really should make this reusable and rename all occurrences to `component-visually-hidden`. Again, sorry - adapted in the issue.

File: Build/Vue/src/components/AllSelector/index.tsx [hunk 3]
```diff
@@ -26,12 +26,14 @@ export default class AllSelector extends Vue {
...
 
     private render(): VNode {
-        // fix me, I'm ugly
         return (
-            <a href='#' onClick={(event: Event) => this.toggleSelect(event, this.listOfIdentifiers)} class='btn btn-sm btn-default'>
-                <i class='fa fa-check-square' v-show={this.isSelected} />
-                <i class='fa fa-square-o' v-show={!this.isSelected} />
-            </a>
+            <span class='component-datatable-checkbox'>
+                <input class='component-datatable-checkbox-input' type='checkbox' id='component-datatable-selectall-123523356'
```
Comment (benjaminkott) [inline Build/Vue/src/components/AllSelector/index.tsx]: `id` should be generic and generated.

File: Build/Vue/src/components/AllSelector/index.tsx [hunk 4]
```diff
@@ -26,12 +26,14 @@ export default class AllSelector extends Vue {
...
+                  value='1' onClick={(event: Event) => this.toggleSelect(event, this.listOfIdentifiers)} />
+                <label class='component-datatable-checkbox-label' for='component-datatable-selectall-123523356'>
```
Comment (benjaminkott) [inline Build/Vue/src/components/AllSelector/index.tsx]: see `id` comment above

File: Build/Vue/src/components/ItemSelector/index.tsx [hunk 5]
```diff
@@ -26,12 +26,15 @@ export default class ItemSelector extends Vue {
...
 
     private render(): VNode {
-        // fix me, I'm ugly
         return (
-          <a href='#' onClick={(event: Event) => this.toggleSelect(event, this.identifier)} className='btn btn-sm btn-default'>
-              <i class='fa fa-check-square' v-show={this.isSelected} />
-              <i class='fa fa-square-o' v-show={!this.isSelected} />
-          </a>
+            <span class='component-datatable-checkbox'>
+                <input class='component-datatable-checkbox-input' type='checkbox' id='component-datatable-select-identifier1234'
```
Comment (benjaminkott) [inline Build/Vue/src/components/ItemSelector/index.tsx]: `id` should be generic and generated.

File: Build/Vue/src/components/ItemSelector/index.tsx [hunk 6]
```diff
@@ -26,12 +26,15 @@ export default class ItemSelector extends Vue {
...
+                  value='1' checked={this.selected.includes(this.identifier)} />
+                <label class='component-datatable-checkbox-label' for='component-datatable-select-identifier1234'
```
Comment (benjaminkott) [inline Build/Vue/src/components/ItemSelector/index.tsx]: see `id` comment above

File: Build/Vue/src/components/List/index.tsx [hunk 7]
```diff
@@ -22,20 +33,74 @@ export default class List extends Vue {
...
         }
 
         return (
-            <table class='table table-striped table-hover'>
-                <thead>
-                    <th><AllSelector listOfIdentifiers={this.items.map((item: any) => {return item.identifier; })}/></th>
-                    {headerColumns}
+            <div class='component-datatable' role='group' aria-labelledby={'component-datatable-' + tstamp}>
+                <table class='component-datatable-table'>
+                    <caption class='component-datatable-caption' id={'component-datatable-' + tstamp}>Files and Folders</caption>
```
Comment (benjaminkott) [inline Build/Vue/src/components/List/index.tsx]: **Files and Folders** should be current folder name.

File: Build/Vue/src/components/List/index.tsx [hunk 8]
```diff
@@ -22,20 +33,74 @@ export default class List extends Vue {
...
+                    <thead class='component-datatable-head'>
```
Comment (benjaminkott) [inline Build/Vue/src/components/List/index.tsx]: indentions?

File: Build/Vue/src/components/List/index.tsx [hunk 9]
```diff
@@ -22,20 +33,74 @@ export default class List extends Vue {
...
+                        </th>
+                        {headerColumns}
+                    </tr>
                 </thead>
-                <tbody>{list}</tbody>
-            </table>
+                <tbody>
+                    {list}
+                </tbody>
+                </table>
```
Comment (benjaminkott) [inline Build/Vue/src/components/List/index.tsx]: indentions?

File: Build/Vue/src/components/List/index.tsx [hunk 10]
```diff
@@ -22,20 +33,74 @@ export default class List extends Vue {
...
+            </div>
+        );
+    }
+
+    private renderHeaderColumn(field: string, icon?: string): VNode {
+        const sortingOrder = this.sorting.order;
+        const sortIcon: any = (this.sorting.field === field) ? (
+            <span class='component-datatable-sorting-icon' role='presentation'>
+              <span class='component-datatable-sorting-icon-asc' v-show={sortingOrder === SortingOrder.ASC} />
```
Comment (benjaminkott) [inline Build/Vue/src/components/List/index.tsx]: Please render SVG/icons here,  `component-datatable-sorting-icon-asc|desc` is not intended here.

File: Build/Vue/src/components/List/index.tsx [hunk 11]
```diff
@@ -22,20 +33,74 @@ export default class List extends Vue {
...
+              <span class='component-datatable-sorting-icon-desc' v-show={sortingOrder === SortingOrder.DESC} />
+            </span>
+        ) : '';
+        const sortText: any = (this.sorting.field === field) ? (
+            <span class='component-datatable-sorting-label component-datatable-visually-hidden'>
+                sort by {TYPO3.lang['List.table.header.' + field]} in {sortingOrder === SortingOrder.ASC
+                ? 'descending' : 'ascending'} order
```
Comment (benjaminkott) [inline Build/Vue/src/components/List/index.tsx]: Language Labels for 
`sort by`, `descending`, `ascending`, `order`

File: Build/Vue/src/components/ListItem/index.tsx [hunk 12]
```diff
@@ -36,53 +36,77 @@ export default class ListItem extends Vue {
...
+                              <img src={this.item.icon} height='16' />
+                          </span>
+                          <span class='component-datatable-link-text'>{val}</span>
+                      </a>
+                  </th>
+                );
+            } else {
+                return (
+                  <th scope='row' data-type='title' data-value={this.item.name} data-property='name' data-label='Filename'>
+                      <a class='component-datatable-link' href={this.item.editMetaUrl} title={'Open ' + this.item.name}>
```
Comment (benjaminkott) [inline Build/Vue/src/components/ListItem/index.tsx]: Language Label for `Open`

File: Build/Vue/public/app.css [hunk 13]
```diff
@@ -100,4 +100,8 @@
 
 .tree-dropzone-allowed {
     background-color: lightblue;
+}
+
+.component-datatable-visually-hidden {
```
[the remark this answers is not in the document]
Comment (NeoBlack) [reply Build/Vue/public/app.css]: nope, remove it if youz add your CSS else it is impossible to work with current state.

[the remark this answers is not in the document]
Comment (benjaminkott) [reply Build/Vue/public/app.css]: i am not sure why this should have an impact on current functionality?

File: Build/Vue/src/components/ItemSelector/index.tsx [hunk 14]
```diff
@@ -26,12 +26,17 @@ export default class ItemSelector extends Vue {
...
-          <a href='#' onClick={(event: Event) => this.toggleSelect(event, this.identifier)} className='btn btn-sm btn-default'>
-              <i class='fa fa-check-square' v-show={this.isSelected} />
-              <i class='fa fa-square-o' v-show={!this.isSelected} />
-          </a>
+            <span class='component-checkbox'>
+                <input class='component-checkbox-input' type='checkbox' id={'component-datatable-select-identifier-' + randomPart}
+                  value='1' checked={this.selected.includes(this.identifier)} />
+                <label class='component-checkbox-label' for={'component-datatable-select-identifier-' + randomPart}
+                    onClick={(event: Event) => this.toggleSelect(event, this.identifier)}>
+                    <span class='component-datatable-visually-hidden'>{label} {this.identifier}</span>
```
Comment (benjaminkott) [inline Build/Vue/src/components/ItemSelector/index.tsx]: please rename `component-datatable-visually-hidden` to `component-visually-hidden`

File: Build/Vue/src/components/List/index.tsx [hunk 15]
```diff
@@ -22,20 +33,93 @@ export default class List extends Vue {
...
-                <thead>
-                    <th><AllSelector listOfIdentifiers={this.items.map((item: any) => {return item.identifier; })}/></th>
-                    {headerColumns}
+            <div class='component-datatable' role='group' aria-labelledby={'component-datatable-' + randomPart}>
+                <table class='component-datatable-table'>
+                    <caption class='component-datatable-caption' id={'component-datatable-' + randomPart}>
+                      {this.current}
+                    </caption>
+                    <thead class='component-datatable-head'>
+                    <tr>
```
Comment (benjaminkott) [inline Build/Vue/src/components/List/index.tsx]: indentions?

File: Build/Vue/src/components/List/index.tsx [hunk 16]
```diff
@@ -22,20 +33,93 @@ export default class List extends Vue {
...
+              </svg>
+            </span>
+        ) : '';
+        const sortInfo = TYPO3.lang['List.table.header.sort_info']
+          .replace('{field}', TYPO3.lang['List.table.header.' + field])
+          .replace('{order}', sortingOrder === SortingOrder.ASC
+            ? TYPO3.lang['List.table.header.descending']
+            : TYPO3.lang['List.table.header.ascending']);
+        const sortText: any = (
+            <span class='component-datatable-sorting-label component-datatable-visually-hidden'>{sortInfo}</span>
```
Comment (benjaminkott) [inline Build/Vue/src/components/List/index.tsx]: please rename `component-datatable-visually-hidden` to `component-visually-hidden`

File: Build/Vue/src/components/List/index.tsx [hunk 17]
```diff
@@ -22,20 +33,93 @@ export default class List extends Vue {
...
+                        </th>
+                        {headerColumns}
+                    </tr>
                 </thead>
-                <tbody>{list}</tbody>
-            </table>
+                <tbody>
+                    {list}
+                </tbody>
+                </table>
```
Comment (benjaminkott) [inline Build/Vue/src/components/List/index.tsx]: Indention

**Pull request merged**